Formula/ethereum.rb

```diff
@@ -21,6 +21,10 @@ class Ethereum < Formula
   depends_on "go" => :build
 
   def install
+    on_linux do
+      # See https://github.com/golang/go/issues/26487
+      ENV.O0
```

Comment:
This will error, since `ENV.O{0,1,2,3}` [has been disabled](https://github.com/Homebrew/brew/blob/04dab8428b4db1bb94bc1284cbaf77322fef9334/Library/Homebrew/extend/ENV/super.rb#L347-L349).

See also https://github.com/Homebrew/homebrew-core/pull/70316.

Reply:
Interesting. CI did not complain though :)

Is there a workaround?

Reply:
That's because CI actually needs to try to call `ENV.O0` in order to encounter the error, but it didn't here.

I'm not aware of a workaround. One would be nice, as it would help the other issue too. However, I'm not sure it would help this formula, as it's not clear `go` inherits from `HOMEBREW_OPTIMIZATION_FLAGS`.

You may be able to pass the flag to go directly since the compiler shim won't eat it. (It will if you try to do it with `clang` or `gcc`.)

Reply:
This report in nix gives more details:
https://github.com/NixOS/nixpkgs/issues/25959

The actual bug is in the vendored https://github.com/karalabe/hid/ package. This was reported here:
https://github.com/karalabe/hid/issues/2

Final conclusion: https://github.com/karalabe/hid/issues/2#issuecomment-309266314

The issue is still there, and using clang to compile go and ethereum should fix it. This is why you do not see the issue on Mac, as everything is compiled with clang. On Linux everything is compiled with gcc.

So either I add a `fails_with gcc` to go and ethereum, or we ship the ENV.O0 fix.

Ideally upstream should fix their bug though.